### PR TITLE
Docs: Tutorials formatting + from scratch connector tutorial cleanup

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

This is a follow-up for airbytehq/airbyte#33822. It formats `tutorials/**/*.md` as an experiment with Prettier, and moves the onboarding "build connnector the hard way" tutorial in it's own subdirectory, and fixes a few inconsistencies.

## Why

I'm not _strongly_ attached to markdown linting and formatting, but I would love to make things nicer for us overall, and I see quite a lot of work that we can do to **make both writing, maintaining, and contributing to Airbyte documentation a better experience**.

## How

- This PR depends on airbytehq/airbyte#33822. That one should be merged first, than this should be rebased.
- Formatting pass is in a single commit, so it's easy to eye through. 

### Content changes in Tutorials

- I have my eyes on Tutorials, specifically airbytehq/airbyte-internal-issues#2382. Moving and cleaning tutorials is something I can do because I'll use the low-code CDK next. Plus, that way I will learn more about our documentation infrastructure. 

### Housekeeping in Docs

- [x] We have quite a lot of outdated deps, and one of them registers on Dependabot security list (Babel). airbytehq/airbyte#33840 
- We should then add a markdown linter to have a list of pages to add front matter fields to, like `summary`. 
- Content cleanup would also be good. Some tutorials still link to assets in `.gitbook`.

### CODE REVIEW

File: docs/connector-development/tutorials/building-a-java-destination.md [hunk 1]
```diff
@@ -2,24 +2,26 @@
 
 ## Summary
 
-This article provides a checklist for how to create a Java destination. Each step in the checklist has a link to a more detailed explanation below.
+This article provides a checklist for how to create a Java destination. Each step in the checklist
+has a link to a more detailed explanation below.
```
Comment (girarda) [inline docs/connector-development/tutorials/building-a-java-destination.md:6]: why the line break?

Comment (natikgadzhi) [reply docs/connector-development/tutorials/building-a-java-destination.md:6]: Most likely a `prettier -w` pass with 100 desired line width. I'm not strongly attached to this approach — I personally like linted markdown to comply to certain width, especially when there are a lot of contibutors, but functionally, same thing.

**Comment by natikgadzhi** [comment]
Caveat: when we merge this, I need to update the engineering onboarding check-list, because it links to the old the-hard-way guide URL.

Also, consider making them linked to the sidebar again.

**Review by girarda — APPROVED** [review]
🚢

**Comment by marcosmarxm** [comment]
@natikgadzhi let's ship this one?

**Pull request merged**